ui/desktop/electron-app/src/index.js

```diff
@@ -184,33 +108,118 @@ app.on('ready', async () => {
       shell.openExternal(url);
     }
   });
 
-  mainWindow.on('unresponsive', () => {
+  win.on('unresponsive', () => {
     console.log(
       'Your Ember app (or other code) has made the window unresponsive.'
     );
   });
 
-  mainWindow.on('responsive', () => {
+  win.on('responsive', () => {
     console.log('The main window has become responsive again.');
   });
 
-  mainWindow.on('closed', () => {
-    mainWindow = null;
+  win.on('closed', () => {
+    closeWindowCB();
+  });
+
+  return win;
+};
+
+let mainWindow = null;
+
+app.on('window-all-closed', () => {
+  if (process.platform !== 'darwin') {
+    app.quit();
+  }
+});
+
+app.on('ready', async () => {
+  // Setup a partition
+  // Must be prefixed with persist: in order to save things like localStorage
+  // data.  Without this, the session is in-memory only.
+  // https://www.electronjs.org/docs/api/session#sessionfrompartitionpartition-options
+  const partition = `persist:${emberAppName}`;
+  const ses = session.fromPartition(partition);
+
+  // Register custom protocol
+  // This file protocol is the exclusive protocol for this application.  All
+  // other protocols are disabled.
+  ses.protocol.registerFileProtocol(emberAppProtocol, (request, callback) => {
+    /* eng-disable PROTOCOL_HANDLER_JS_CHECK */
+    const isDir = request.url.endsWith('/');
+    const absolutePath = request.url.substr(emberAppURL.length);
+    const normalizedPath = isDir
+      ? path.normalize(`${emberAppDir}/index.html`)
+      : path.normalize(`${emberAppDir}${absolutePath}`);
+
+    if (isDev) console.log('[serving]', request.url);
+
+    callback({ path: normalizedPath });
+  });
+
+  // Disallow all permissions requests,
+  // per Electronegativity PERMISSION_REQUEST_HANDLER_GLOBAL_CHECK
+  ses.setPermissionRequestHandler((webContents, permission, callback) => {
+    /* eng-disable PERMISSION_REQUEST_HANDLER_JS_CHECK */
+    return callback(false);
   });
+
+  // Setup content security policy
+  ses.webRequest.onHeadersReceived((details, callback) => {
+    callback({
+      // See `content-security-policy.js` to learn how the CSP is generated.
+      responseHeaders: {
+        ...details.responseHeaders,
+        'Content-Security-Policy': [generateCSPHeader()],
+      },
+    });
+  });
+
+  if (isDev) {
+    try {
+      require('devtron').install();
+    } catch (err) {
+      console.log('Failed to install Devtron: ', err);
+    }
+    try {
+      await installExtension(EMBER_INSPECTOR);
+    } catch (err) {
+      console.log('Failed to install Ember Inspector: ', err);
+    }
+  }
+
+  // Configure dev tools menu
+  const menuTemplate = Menu.buildFromTemplate(menu.generateMenuTemplate());
+  if (isDev) {
+    const view = menuTemplate.getMenuItemById('view');
+    view.submenu.append(new MenuItem({ role: 'toggleDevTools' }));
+  }
+  Menu.setApplicationMenu(menuTemplate);
+
+  // Close window callback
+  const closeWindowCB = () => {
```

Comment:
**IMPORTANT:**
I created this callback to keep the same behaviour than before, when `mainWindow` receives the close event, it equals `mainWindow` to `null`.